### ISSUE TITLE
[BUG] - Uploading an image crashes Mealie instantly

### ISSUE THREAD

**Issue by felixschndr** [mealie-recipes/mealie#3848]
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

If I upload an image to a recipe Mealie instantly crashes. 

This only seems to happen with images from my camera. Uploading an Android screenshot works without any issues. I guess there is some metadata in the images from my camera or the resolution brings Mealie to a halt.

### Steps to Reproduce

1. Create a recipe
2. Add this image to the recipe: 
![foto_no_exif (1)](https://github.com/mealie-recipes/mealie/assets/45742226/03c8146f-4e50-4605-ba24-1167bf832da6)
3. See mealie crash

### Please provide relevant logs

These are the only logs (level set to `DEBUG`):
```text
mealie_database   | 2024-07-04 20:44:55.714 UTC [636] LOG:  unexpected EOF on client connection with an open transaction
mealie_app exited with code 0
```

### Mealie Version

Version: v1.10.1 (happend with 1.8.1 as well)
Build: [d639d168fa7c39ffb802d46ba05e4ec4de33faf9](https://github.com/mealie-recipes/mealie/commit/d639d168fa7c39ffb802d46ba05e4ec4de33faf9)

### Deployment

Docker (Linux)

**Comment by felixschndr** [comment]
Edit: It was not a problem with the EXIF metadata of the file. I used an online tool to remove it and the bug still occured. It seems to be a problem with the size or resolution. The original image has a resolution of 4000x3000. After I cropped it a little (to 3795x2065) the bug does not appear any more

**Comment by boc-the-git** [comment]
Please share your docker-compose file.

Can this be replicated on an installation using our sample postgres docker-compose?

**Comment by boc-the-git** [comment]
You've said this can be replicated on the demo, what are the exact steps for that?
This is the demo instance:
![image](https://github.com/user-attachments/assets/3e3450e6-11ab-4942-9093-fe450e13bdef)

**Comment by felixschndr** [comment]
> Please share your docker-compose file.

```yml
services:
  mealie:
    image: ghcr.io/mealie-recipes/mealie
    container_name: mealie_app
    ports:
        - 12000:9000
    deploy:
      resources:
        limits:
          memory: 400M
    volumes:
      - ./data/mealie:/app/data/
    environment:
      ALLOW_SIGNUP: "false"
      PUID: ${ENV_PUID}
      PGID: ${ENV_PGID}
      TZ: ${TIMEZONE}
      API_DOCS: "false"
      MAX_WORKERS: 1
      WEB_CONCURRENCY: 1
      DB_ENGINE: postgres
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_SERVER: mealie_database
      POSTGRES_PORT: 5432
      POSTGRES_DB: ${POSTGRES_DB}
      SMTP_HOST: ${SMTP_HOST}
      SMTP_PORT: ${SMTP_PORT}
      SMTP_FROM_NAME: ${SMTP_FROM_NAME}
      SMTP_AUTH_STRATEGY: ${SMTP_AUTH_STRATEGY}
      SMTP_FROM_EMAIL: ${SMTP_FROM_EMAIL}
      SMTP_USER: ${SMTP_USER}
      SMTP_PASSWORD: ${SMTP_PASSWORD}
    restart: always
    depends_on:
      - database
  database:
    container_name: mealie_database
    image: postgres:15
    volumes:
      - ./data/postgres/postgres:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    restart: always
    healthcheck:
      test: pg_isready -U $$POSTGRES_USER -d $$POSTGRES_DB
      interval: 5s
      timeout: 5s
      retries: 5
```

----

> You've said this can be replicated on the demo, what are the exact steps for that?

I just tried this on my personal mealie instance. The steps are as follows
1. Download the image from above (normal right click and save)
2. Create a new recipe
3. Add the image (via upload)
4. See mealie crash in the logs

This is with the newest available docker version (Version `v1.10.2`, Build [e344f3f1e6672d673f6e0899a5cef28aa8beb2f0](https://github.com/mealie-recipes/mealie/commit/e344f3f1e6672d673f6e0899a5cef28aa8beb2f0).

I just tried it again in the demo instance and was **not** able to reproduce it. However I am pretty sure I was able to reproduce it in the demo instance in the past.

**Comment by felixschndr** [comment]
I made you a screen recording (maybe it helps). You can find it [here](https://drive.google.com/file/d/1uZ0iX2tLeX3rXxuPcQePntvuoY4Xx9vO/view?usp=sharing)

**Comment by boc-the-git** [comment]
@felixschndr can you post your compose file with code formatting. 
```
It
Should
Look
Something like this
```

It's not really readable in its current state. 

One initial suggestion, try increasing the memory limit, at least to 1000

**Comment by felixschndr** [comment]
> can you post your compose file with code formatting.

Yes, of course. I tried doing a summary but that messed up the formatting. I changed the original post.

----

> try increasing the memory limit, at least to 1000

That fixed the issue. Thanks. However I am wondering about two things
1. Why does Mealie need that much RAM to work with the image?
2. Maybe Mealie could throw some kind of exception when it recognizes it runs out of memory. I don't know if this is possible. There should be a `MemoryError` https://docs.python.org/3/library/exceptions.html#MemoryError
